ocaml/xapi/repository_helpers.ml

```diff
@@ -1079,3 +1079,24 @@ let with_access_token ~token ~token_id f =
   | _ ->
       let msg = Printf.sprintf "%s: The token or token_id is empty" __LOC__ in
       raise Api_errors.(Server_error (internal_error, [msg]))
+
+let sort_in_assoc x =
+  Yojson.Basic.Util.to_assoc x
+  |> List.sort (fun (k1, _) (k2, _) -> String.compare k1 k2)
+  |> fun x -> `Assoc x
+
+let prune_updateinfo_for_livepatches livepatches updateinfo =
+  let updateinfo' = Yojson.Basic.Util.to_assoc updateinfo in
+  match List.assoc_opt "livepatches" updateinfo' with
+  | None ->
+      updateinfo
+  | Some l ->
+      let livepatches' = List.map sort_in_assoc livepatches in
+      let l' =
+        Yojson.Basic.Util.to_list l
+        |> List.filter (fun x -> List.mem (sort_in_assoc x) livepatches')
```

Comment:
Could this be done using Set intersection and Set difference operations instead of operations on lists? Might be more natural to express it that way, and Sets would guarantee uniqueness of entries, and a good sorted order when converting back to lists.